src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -366,42 +371,33 @@ def test_parallellooptrans_validate_dependencies(fortran_reader):
     del loops[1].ancestor(Routine).symbol_table._symbols['zws']
     omplooptrans.validate(loops[1])
 
-    # Reductions have dependencies but these are accepted because it
-    # can be manage with the appropriate clause
-    psyir = fortran_reader.psyir_from_source('''
-    subroutine my_subroutine()
-        integer :: ji, jj, jk, jpkm1, jpjm1, jpim1
-        real, dimension(10, 10, 10) :: zwt
-        real :: total
+    # Reductions also indicate a data dependency that needs to be handled, so
+    # we don't permit the parallelisation of the loop (until we support
+    # reduction clauses)
+    loops = create_loops('''
         do jk = 2, jpkm1, 1
           do jj = 2, jpjm1, 1
             do ji = 2, jpim1, 1
               total = total + zwt(ji,jj,jk)
             enddo
           enddo
-        enddo
-    end subroutine''')
-    loops = psyir.walk(Loop)
-    assert not DependencyTools().can_loop_be_parallelised(
-                    loops[0], only_nested_loops=False)
-    omplooptrans.validate(loops[0])
+        enddo''')
+    with pytest.raises(TransformationError) as err:
```

Comment:
I see this works as I'd expect so I don't know what I'm missing in the code where you've added the READWRITE access?

Reply:
See above.